### ISSUE TITLE
Install specific version of BigBlueButton via apt - general issues with BigBlueButton and apt

### ISSUE THREAD

**Issue by BrutalBirdie** [bigbluebutton/bigbluebutton#10914]
TL;DR
I want to install BigBlueButton Version 2.2.28 instead of 2.2.29 via apt but simply cant.

First of all yes I have read these issues:

- https://github.com/bigbluebutton/bigbluebutton/issues/10746#issuecomment-719165922
- https://github.com/bigbluebutton/bigbluebutton/issues/9508#issuecomment-652897005

Now let me explain myself.
I do not setup BigBlueButton via `bbb-install.sh`.
I use Ansible (which uses apt) - a deviation/fork of [n0emis/ansible-role-bigbluebutton](https://github.com/n0emis/ansible-role-bigbluebutton/issues) which is recommended way for bigger instances - See https://docs.bigbluebutton.org/2.2/install.html#ansible

Via apt the only version of bigbluebutton is:
```bash
# apt-cache madison bigbluebutton
bigbluebutton |  1:2.2.0-5 | https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
```
File content of: `/etc/apt/sources.list.d/ubuntu_bigbluebutton_org_xenial_22.list`
```
deb https://ubuntu.bigbluebutton.org/xenial-22/ bigbluebutton-xenial main
```

On another server which is older then 22 Days (before BigBlueButton version 2.2.29), the same apt setting are present but the bigbluebutton version is `2.2.28`
which also implies another issue.
I can not update BigBlueButton via apt - see https://docs.bigbluebutton.org/2.2/install.html#1-update-your-server

I run:
```
apt-get update
apt-get dist-upgrade --dry-run

The following packages will be upgraded:
  bbb-apps-akka bbb-config bbb-html5 bbb-web bbb-webrtc-sfu

apt serach bigbluebutton
# and I get
bigbluebutton/unknown,now 1:2.2.0-5 amd64 [installed]
  Open source web conferencing platform (bbb)
```
Also got reported here https://github.com/n0emis/ansible-role-bigbluebutton/issues/65

So basically I am forced to use `bbb-install.sh` if I wish to downgrade or install another version? Is this correct? Is this intended?

I must be missing something?

**Comment by uka-support** [comment]
Please see **https://docs.bigbluebutton.org/2.2/getting-help.html#help-installing-or-running-bigbluebutton** on how to get help. This is an issue tracker, not a support forum. 

The support forum for BBB is here: https://groups.google.com/forum/#!forum/bigbluebutton-setup

It paackage version numbers are unfortunetly pretty useless. (even ```bbb-conf --version``` is rather useless, Use ```bbb-conf --check``` to get useful ones. This is a working BBB 2.2.29 installation:

```
# apt-cache madison bigbluebutton
bigbluebutton |  1:2.2.0-5 | https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
```

As far as I know ```bbb-install.sh``` is the primary supported method of installation. It get's all the focus. I'd suggest to either take a look at ```bbb-install.sh``` and/or perform a sample installation using ```bbb-install.sh```, then look at all the installed packages.

**Comment by lonesomewalker** [comment]
If you run a playbook, you should already know the solution... Bash is free to use.

**Comment by ffdixon** [comment]
We would recommend you install 2.2.29 due to some security updates (see https://github.com/bigbluebutton/bigbluebutton/releases/tag/v2.2.29).

Curious why you want to install 2.2.28?

**Comment by BrutalBirdie** [comment]
> If you run a playbook, you should already know the solution... Bash is free to use.

Yea.
But feels wrong.
I know that with the `bbb-install.sh` there are no problems.

Still the problem remains.



> We would recommend you install 2.2.29 due to some security updates (see https://github.com/bigbluebutton/bigbluebutton/releases/tag/v2.2.29).
> 
> Curious why you want to install 2.2.28?

I stopped arguing with customers about that. I tell them,this is not good and should not be done this way.
Customer:"But I want Version 0.0.1-dev-bugged-alpha-foobar since it could do a backflip when I pressed the wrong button."

**Comment by lonesomewalker** [comment]
> Yea.
> But feels wrong.
> I know that with the `bbb-install.sh` there are no problems.

:D but there are other wrong things in ansible.
We also tried it in the beginning, since it saves A LOT of time.
But actually we are running the bbb-install.sh as raw. Gives other headaches, but in general no big issues (even with larger setups). And you can install downgrades :P

> I stopped arguing with customers about that. I tell them,this is not good and should not be done this way.
> Customer:"But I want Version 0.0.1-dev-bugged-alpha-foobar since it could do a backflip when I pressed the wrong button."

We also offer our customers this service, but this leads down to special settings and using different playbooks.
Yes, looks ugly, but works ;-)

**Comment by BrutalBirdie** [comment]
I resolved my issue by now installed bbb via the recommended `bbb-install.sh` script BUT as an Ansible tasks where the parameters are filled by variables.

Still this does not solve the reason why I created this issue.

The official 'README' https://docs.bigbluebutton.org/2.2/install.html#1-update-your-server states to update via apt, which is not working.

**Comment by lonesomewalker** [comment]
May you post your playbook?
We maintain several thousand servers in this way.
Usually there are no issues, rarely Let's Encrypt buggs around.
